### ISSUE TITLE
business closures

### ISSUE THREAD

**Issue by tornadoblue** [awfulvizz/reservoir_coffee#2]
The cafe will be closed over the holiday periods and so it would be useful to serve this info from an API and update the code to reflect when the cafe is open if the day falls within the closure period.

The web host has a Hasura GraphQL engine/endpoint setup for this to simply the administration and data queries.  I'm happy to take this on but probably won't get to it until the 20th or so.

[business_hours.txt](https://github.com/awfulvizz/reservoir_coffee/files/7665117/business_hours.txt)

**Comment by JordanAbel** [comment]
Which web host did you use? I'm not too familiar with Hasura GraphQL but I can try to figure it out. Is it essentially just a different way to make calls to an API rather than using a RESTful API?

**Comment by JordanAbel** [comment]
Also, I'm not sure if you can add me as a user but I am unable to deploy
<img width="414" alt="Screen Shot 2021-12-06 at 8 24 21 PM" src="https://user-images.githubusercontent.com/93851341/144966108-5b3afaa1-10bf-457d-9976-6f981b9c70f1.png">

**Comment by tornadoblue** [comment]
> Which web host did you use? I'm not too familiar with Hasura GraphQL but I can try to figure it out. Is it essentially just a different way to make calls to an API rather than using a RESTful API?

Yes, but you can also define your GraphQL queries as a REST endpoint so have best of both worlds.   I've added the API documentation for Docker and Hasura GraphQL, will do a PR now

**Comment by tornadoblue** [comment]
I'm using an Ubuntu VPS hosted by linode.com  -  I've setup docker & docker-compose and using an nginx reverse proxy for exposing port 8080 as https://<domain>/api/rest 

For pushing your build out to production, I'll need to update permissions and groups first but shouldn't be too hard to configure.

**Comment by tornadoblue** [comment]
Also it should default to the prod REST API if you don't configure a local Hasura GraphQL endpoint so you don't have to setup the GraphQL stuff if you don't want to get your hands dirty with that.

**Comment by JordanAbel** [comment]
Okay, I'll look into it, thanks for your help. I've only used Django for creating an API so this will be fun, I'm excited to learn about it. We are also using docker and nginx where I am doing my internship but I haven't been involved enough to know too much about how it is set up.

**Comment by JordanAbel** [comment]
Im curious why we want to use an API for this feature though, can't we achieve this by simply checking the date and render (...) accordingly?

**Comment by JordanAbel** [comment]
Say we have `BUISNESS_HOURS_DATA.jsx`

Couldn't we then do something like
`export const BUISNESS_HOURS = [
      {
        "id": 1,
        "day": "Sunday",
        "start": null,
        "end": null
      } ....
    ]`

**Comment by tornadoblue** [comment]
The API will be used for updating the hours and closure dates as well, so management can use a simple admin portal to edit which days the cafe is closed, or update the core hours without having to push out a new version of the app.

**Comment by JordanAbel** [comment]
Ahh, okay I see. I thought about creating an API for the site when Kristine was saying she didn't want the prices for menu items to be displayed because of constant updates if she ever wanted to change the price of any items. 

Sorry for all of the questions, hopefully I'm not taking too much of your time

**Comment by tornadoblue** [comment]
No worries at all, happy to help out in any way!   Will email some details your way for deploying the builds.

**Comment by tornadoblue** [comment]
Added a business_status view in Hasura and included the result in the business_info API result.  It uses checks for any active business closures and whether the current_date time is between the current day's open/closed times to determine if the status is "open" or "closed".

The structure currently looks as below:

`{"business_status":[{"current_date":"2021-12-07T23:21:19.376159","today":"Tuesday","status":"closed","opens":"08:00:00-08","closes":"16:00:00-08","closed_reason":null}]}`

**Comment by tornadoblue** [comment]
Was a problem with the date formatting when viewing from Canada due to JS Date handling of the timestamps.  Rewritten the date parser so that should be fixed.  Should display closures from 24 Dec to 02 Jan on the location page now on test site:
https://test.reservoircoffee.ca/location  - will PR when confirmed showing correctly

**Comment by tornadoblue** [comment]
Pushed out a version to the live site from my fork, holidays can be added/editing via the hasura admin but should automatically be removed from the query after the closure periods.

**Comment by JordanAbel** [comment]
Sounds good. Sorry for the late reply, I've been fairly busy with family over the holidays and haven't had much of a chance to use my computer.

**Comment by tornadoblue** [comment]
All good, enjoy the xmas holidays with your family!   I will be over there
in a couple months time, we have also been pretty hectic busy getting ready
for the move to Canada for the next couple years but pretty excited for the
big change.
